plugins/field-colour/README.md

```diff
@@ -64,12 +72,55 @@ Blockly.defineBlocksWithJsonArray([
 ]);
 ```
 
+### Blocks
+
+This package also provides four blocks related to the colour field. Each block has generators in JavaScript, Python, PHP, Lua, and Dart.
+- "colour_blend" takes in two colours and a ratio and outputs a single colour.
+- "colour_picker" is a simple block with just the colour field and an output.
+- "colour_random" generates a random colour.
+- "colour_rgb" generates a colour based on red, green, and blue values.
+
+You can install all four blocks by calling `FieldColour.installAllBlocks()`. This will install the blocks and all of their dependencies, including the colour field. You can choose which language generators to install by adding or removing items in the map of generators passed to `installAllBlocks`.
+
+```js
+import { javascriptGenerator } from 'blockly/javascript';
+import { dartGenerator } from 'blockly/dart';
+import { phpGenerator } from 'blockly/php';
+import { pythonGenerator } from 'blockly/python';
+import { luaGenerator } from 'blockly/lua';
+import { FieldColour } from '@blockly/field-colour';
+
+// Installs all four blocks, the colour field, and all of the language generators.
+FieldColour.installAllBlocks({
```

Comment:
Open question: where should I document the `generators` parameter in more detail? Where should the type live? Currently it's in `blocks/generatorUtils.ts` and needs to be duplicated into every plugin where I define blocks.

Reply:
My concern is that I will also need to declare this type in `field_angle` and `field_multilinetextinput` when I update those, because there's no shared place for this type to live.

Reply:
Are you proposing to move the declaration to Blockly itself?  I can see an argument in favour (it would be useful to have in a common location, and why not add it to the thing that all the client code will be importing anyway?), but on balance I am against adding a type to core that's not used anywhere in core.

There might be an argument for adding it to a utility package of field and block definition support code, but so far I think it's just this one type, and the type is straightforward enough that I don't think it's worth creating a separate package for it.  DRY is a useful principle but not IMHO a compelling reason to incur the expense of an extra dependency for this.

If we end up with a bunch of declarations that are actually complicated enough to be worth factoring out then I would think the utility package would be the way to go.

Of course your original question pertains to _documenting_ the type.  To that I feel like the examples usages in the readme are sufficient, especially in conjunction with the type's declaration and JSDoc comment.